### ISSUE TITLE
Domain wide delegation using Service Accounts and JSON 

### ISSUE THREAD

**Issue by lesv** [googleapis/google-api-java-client#1037]
This is needed for google/google-auth-library-java#90 and `google-api-client` (creating an issue for both projects):

User's current way:
```java
GoogleCredential googleCredential = new GoogleCredential.Builder()
     .setTransport(TRANSPORT).setJsonFactory(JSON_FACTORY)
     .setServiceAccountId(emailAddress)
     .setServiceAccountPrivateKeyFromP12File(p12File)
     .setServiceAccountScopes(scopes)
     .setServiceAccountUser(user).build();
```

Would like: to do:
```java
GoogleCredential googleCredential = 
    GoogleCredential.fromStream(jsonInputStream, TRANSPORT, JSON_FACTORY).createScoped(Collections.singleton(Oauth2Scopes.USERINFO_EMAIL))
```

Python you just do:
```python
credentials = credentials.create_delegated(user)
```

**Comment by diegogarciateba** [comment]
Hi,

Is there any timeline to include this functionality in the library?

Thanks in advance!

**Comment by gartediego** [comment]
Hi again,

Regarding one more time the comment of @lesv, what we need would be something like that:

```java
GoogleCredential googleCredential = GoogleCredential.fromStream(jsonInputStream, TRANSPORT, JSON_FACTORY)
      .createScoped(Collections.singleton(Oauth2Scopes.USERINFO_EMAIL))
      .createDelegated(userEmail);
```

This would be similar to the piece of code we are currently using to generate user tokens with specific scopes.

```java
GoogleCredential googleCredential = new GoogleCredential.Builder()
     .setTransport(TRANSPORT).setJsonFactory(JSON_FACTORY)
     .setServiceAccountId(emailAddress)
     .setServiceAccountPrivateKeyFromP12File(p12File)
     .setServiceAccountScopes(scopes)
     .setServiceAccountUser(user).build();
```

Could it be possible to include this functionality in the library?

Thanks!

**Comment by chingor13** [comment]
We just accepted #1079 which adds the `createDelegated` method. It will be released in 1.26.0.
